	Required for gcc-3.4 and probaly other old versions to avoid
	  error: base `graphite2::Face' with only non-default constructor in
          class without a constructor
	and the same for Font, Segment.

diff -ur graphite2-1.2.0.orig/src/inc/Face.h graphite2-1.2.0/src/inc/Face.h
--- graphite2-1.2.0.orig/src/inc/Face.h	2012-09-21 10:39:40.000000000 +0200
+++ graphite2-1.2.0/src/inc/Face.h	2012-12-04 12:15:58.000000000 +0100
@@ -89,6 +89,9 @@
 
     CLASS_NEW_DELETE;
 private:
+    // do not instantiate
+    Face();
+
     SillMap                 m_Sill;
     gr_face_ops             m_ops;
     const void            * m_appFaceHandle;    // non-NULL
diff -ur graphite2-1.2.0.orig/src/inc/Font.h graphite2-1.2.0/src/inc/Font.h
--- graphite2-1.2.0.orig/src/inc/Font.h	2012-09-21 10:39:40.000000000 +0200
+++ graphite2-1.2.0/src/inc/Font.h	2012-12-04 12:13:14.000000000 +0100
@@ -54,6 +54,9 @@
     float               m_scale;      // scales from design units to ppm
     bool                m_hinted;
 
+    // do not instantiate
+    Font();
+
     Font(const Font&);
     Font& operator=(const Font&);
 };
diff -ur graphite2-1.2.0.orig/src/inc/Segment.h graphite2-1.2.0/src/inc/Segment.h
--- graphite2-1.2.0.orig/src/inc/Segment.h	2012-09-21 10:39:40.000000000 +0200
+++ graphite2-1.2.0/src/inc/Segment.h	2012-12-04 12:15:53.000000000 +0100
@@ -148,6 +148,9 @@
     float justify(Slot *pSlot, const Font *font, float width, enum justFlags flags, Slot *pFirst, Slot *pLast);
   
 private:
+    // do not instantiate
+    Segment();
+
     Rect            m_bbox;             // ink box of the segment
     Position        m_advance;          // whole segment advance
     SlotRope        m_slots;            // std::vector of slot buffers
